libs/community/langchain_community/tools/playwright/base.py

```diff
@@ -31,10 +31,14 @@ def lazy_import_playwright_browsers() -> Tuple[Type[AsyncBrowser], Type[SyncBrow
         guard_import(module_name="playwright.sync_api").Browser,
     )
 
+class BaseBrowserToolInput(BaseModel):
+    """Base class for browser tool input when tool has no arguments."""
 
 class BaseBrowserTool(BaseTool):
     """Base class for browser tools."""
 
+    args_schema: Type[BaseModel] = BaseBrowserToolInput
```

Comment:
@colriot why do we need to define a default schema on the base class? Ideally sub-classes would be forced to define the schema? Could we remove it here and add an appropriate schema in each tool that uses it?

I think definition here would make sense if this was called "ParameterlessBaseBrowserTool", but I think it's meant to capture any sort of tool and the main role of the base abstraction is to make sure that it accepts the browser attributes properly

Reply:
@eyurtsev because my suggestion was to have a default NoArgs schema applied. And only tools that actually require non-empty params will need to explicitly override it. I.e. I don't see the reason for subclasses to define their own schema if it's NoArgs (same as it was before the fix, but actually working).

I have a few suggested options:
- Rename `BaseBrowserToolInput` to `NoArgsToolInput` (or smth along the lines) and have it used as default schema value in `BaseBrowserTool`
- Rename input, but remove default value from `BaseBrowserTool` and explicitly use it in every tool
- Keep your suggestion, but then we'll have a few empty classes of no real use, that have identical purpose